### PR TITLE
Tabletmanager tests cases migrated from Python

### PR DESCRIPTION
**Pull request by ajeetj**
Adding/Fixing test cases that were flaky before.
Signed-off-by: Ajeet jain <ajeet@planetscale.com>

### CODE REVIEW

**Comment by ajeetj** [comment]
@deepthi Need your help to debug why `TestIgnoreHealthError` is failing here.

The structure of the test is like this: 
We will have a `NOT_SERVING` tablet. 
We then execute vtctl cmd `IgnoreHealthError zone1-slaveuid .*no slave status.* `
Ideally, this should ignore the slave status and show us that the tablet is `SERVING` 
Which is not working for me. 

The difference between Python and GO is:
In Go, I have shutdown the mysql forcing tablet to go in NOT_SERVING mode. I have also tried it by shutting down the master.
In Python we are only starting a replica. And without a master the replica will remain in NOT_SERVING mode. 

 Code pointer: https://github.com/planetscale/vitess/blob/tal_tabletmanager_4/go/test/endtoend/tabletmanager/tablet_health_test.go#L249

**Comment by deepthi** [comment]
> @deepthi Need your help to debug why `TestIgnoreHealthError` is failing here.
> 
> The structure of the test is like this:
> We will have a `NOT_SERVING` tablet.
> We then execute vtctl cmd `IgnoreHealthError zone1-slaveuid .*no slave status.* `
> Ideally, this should ignore the slave status and show us that the tablet is `SERVING`
> Which is not working for me.
> 
> The difference between Python and GO is:
> In Go, I have shutdown the mysql forcing tablet to go in NOT_SERVING mode. I have also tried it by shutting down the master.
> In Python we are only starting a replica. And without a master the replica will remain in NOT_SERVING mode.
> 
> Code pointer: https://github.com/planetscale/vitess/blob/tal_tabletmanager_4/go/test/endtoend/tabletmanager/tablet_health_test.go#L249

If mysql is down, you will get a different error (probably CRServerGone or CRServerLost), not `no slave status`.
https://github.com/vitessio/vitess/blob/master/go/vt/mysqlctl/replication.go#L219
https://github.com/vitessio/vitess/blob/1469afb171a9dc6b1f7e6652ec9b890b1277f51d/go/vt/mysqlctl/query.go#L34

**Comment by ajeetj** [comment]
> > @deepthi Need your help to debug why `TestIgnoreHealthError` is failing here.
> > The structure of the test is like this:
> > We will have a `NOT_SERVING` tablet.
> > We then execute vtctl cmd `IgnoreHealthError zone1-slaveuid .*no slave status.* `
> > Ideally, this should ignore the slave status and show us that the tablet is `SERVING`
> > Which is not working for me.
> > The difference between Python and GO is:
> > In Go, I have shutdown the mysql forcing tablet to go in NOT_SERVING mode. I have also tried it by shutting down the master.
> > In Python we are only starting a replica. And without a master the replica will remain in NOT_SERVING mode.
> > Code pointer: https://github.com/planetscale/vitess/blob/tal_tabletmanager_4/go/test/endtoend/tabletmanager/tablet_health_test.go#L249
> 
> If mysql is down, you will get a different error (probably CRServerGone or CRServerLost), not `no slave status`.
> https://github.com/vitessio/vitess/blob/master/go/vt/mysqlctl/replication.go#L219
> https://github.com/vitessio/vitess/blob/master/go/vt/mysqlctl/query.go#L34

Got it. I have replicated the Python test case scenario and now the tests are passing.

File: go/test/endtoend/tabletmanager/tablet_health_test.go [hunk 1]
```diff
@@ -244,55 +246,205 @@ func TestHealthCheckDrainedStateDoesNotShutdownQueryService(t *testing.T) {
...
+	err = tablet.VttabletProcess.WaitForTabletType("NOT_SERVING")
+	assert.Nil(t, err)
+	checkHealth(t, tablet.HTTPPort, true)
+
+	// Tear down custom processes
+	killTablets(t, tablet)
 }
 
-func TestIgnoreHealthError(t *testing.T) {
+func TestNoMysqlHealthCheck(t *testing.T) {
```
Comment (deepthi) [inline go/test/endtoend/tabletmanager/tablet_health_test.go]: Does this mean that we are not testing the IgnoreHealthError functionality? Is there a different test case that covers it?

Comment (ajeetj) [reply go/test/endtoend/tabletmanager/tablet_health_test.go]: That's just github's view. They usually mess up if the same signature method is added up/down. 
The `TestIgnoreHealthError` is still present at line 249.

Comment (deepthi) [reply go/test/endtoend/tabletmanager/tablet_health_test.go]: Ah ok

**Pull request merged**